packages/workspace/src/tasks-runner/cache.ts

```diff
@@ -97,12 +103,17 @@ export class Cache {
     writeFileSync(join(td, 'terminalOutput'), terminalOutput);
 
     mkdirSync(join(td, 'outputs'));
-    folders.forEach((f) => {
-      const srcDir = join(this.root, f);
-      if (existsSync(srcDir)) {
-        const cachedDir = join(td, 'outputs', f);
-        fsExtra.ensureDirSync(cachedDir);
-        fsExtra.copySync(srcDir, cachedDir);
+    outputs.forEach((f) => {
+      const src = join(this.root, f);
+      if (existsSync(src)) {
+        const cached = join(td, 'outputs', f);
+        if (lstatSync(src).isDirectory()) {
+          fsExtra.ensureDirSync(cached);
+        } else {
+          fsExtra.ensureFileSync(cached);
```

Comment:
Should not we instead ensureDir for the parent of cached? Will ensureFileSync create an empty file?

Reply:
`ensureFileSync` will already ensure that the parent of `cached` is created, from the extra-fs [docs](https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/ensureFile-sync.md): "Ensures that the file exists. If the file that is requested to be created is in directories that do not exist, these directories are created."

Yes this does create an empty file, but then that file is overwritten in the copy step down below this (overwrite being the default behavior of copy).

Technically I think you're right you'd want to go one directory up from this file, and create that, but I didn't want to have to worry about parsing directory incorrectly here, but I can give this another go, if you'd like.

Reply:
Okay, I think I figured out a good approach to get that parent directory without having to do parsing, so now this either ensures the dir exists in the case that `src` is a dir, or ensures the parent dir exists in the case that `src` is a file. Let me know if this was what you were thinking.